doc/Terminologies.rst

```diff
@@ -0,0 +1,93 @@
+Terminologies and Definitions
```

Comment:
Please list as more as possible terminologies, and we can delete them if they are not needed.

Reply:
Sure